### DISCUSSION TITLE
Security concerns when running grafana agent as agent as root and inside host pid namespace

### DISCUSSION THREAD

**Discussion by megalucio** [grafana/pyroscope#3067] [Q&A]
Have some concerns regarding having to run agent as root and inside host pid namespace in production environments, as stated as requirement [here](https://github.com/grafana/pyroscope/discussions/3067). Is there any way we can work around this?

Would it be possible to narrow down this to some specific privileges and/or capabilities that will be needed in order to assign the minimum amount of permissions here.

Thanks

**Comment by korniltsev** [comment]
With this fix https://github.com/grafana/pyroscope/commit/d58b751dc0efe2583dcddbddb9c01aa2bf7312ab You can now run agent NOT in root pid namespace.

If you can not run agent as privileged root user, you would likely need `CAP_BPF` and `CAP_PERFMON` and kernel >= 5.8 and also modify seccomp policies (probably to allow ptrace and access to `/proc/${pid}/mem` and `/proc/${pid}/root` ) .

In my opinion, this set of capabilities is no different from privileged root user so I can not recommend narrowing down prigileges.
